src/machines/timer_state_machine.rs

```diff
@@ -271,13 +272,8 @@ mod test {
             8: EVENT_TYPE_WORKFLOW_TASK_STARTED
 
             We have two versions of this test, one which processes the history in two calls,
-            and one which replays all of it in one go. The former will run the event loop three
-            times total, and the latter two.
-
-            There are two workflow tasks, so it seems we should only loop two times, but the reason
-            for the extra iteration in the incremental version is that we need to "wait" for the
-            timer to fire. In the all-in-one-go test, the timer is created and resolved in the same
-            task, hence no extra loop.
+            and one which replays all of it in one go. Both versions must produce the same number
+            of activations.
```

Comment:
I mentioned in the SDK channel that Java in fact has this same behavior in the timer test, where the event loop runs three times - but that doesn't necessarily mean there are three activations. I will look into this more and confirm that there are only two activations, but three instances of driving the machines.

Reply:
@mfateev I did confirm this was true, here and in Java, would appreciate your take on this and we may want to restore some of the original comment.